lib/membrane_h264_ffmpeg/parser.ex

```diff
@@ -269,30 +270,19 @@ defmodule Membrane.H264.FFmpeg.Parser do
   end
 
   @impl true
-  def handle_stream_format(:input, %Membrane.H264.RemoteStream{}, ctx, _state)
+  def handle_stream_format(:input, %Membrane.H264.RemoteStream{} = stream_format, ctx, state)
       when ctx.pads.input.start_of_stream? do
-    raise "Cannot handle Membrane.H264.RemoteStream format after the stream has started"
+    Membrane.Logger.debug("Disposing old parser due to stream format change")
+
+    {[], state} = handle_init(%{}, state.original_opts)
```

Comment:
Storing the opts seems like a hack rather than a solution. I'd instead create a private helper (maybe `reset_state`) with the code from `handle_init` and call it both here and there.